# **UAP PRAKTIKUM AI**
Nama : Dede Juniar Putra

NPM : 2217051015

Kelas : B

***TEMA 3***

**Menentukan Penilaian Kinerja Karyawan**

Anda adalah seorang manajer dari sebuah perusahaan, saat ini anda
ingin menentukan karyawan terbaik dalam satu tahun terakhir
berdasarkan kinerja mereka. Menggunakan metode AHP, bagaimana
anda akan menentukan karyawan terbaik anda? Gunakan kriteria dan
seleksi karyawan yang kredibel menurut anda.

Kita akan menggunakan tiga alternatif karyawan dan kriteria sebagai berikut:
1. Nada
2. Sheva
3. Dede
4. Firda

Komunikasi(K), Kerja Sama Tim(KST), Kreativitas(KV), Ketepatan Waktu(KW)


---

**Nada**


*   K = 8
*   KST = 7
*   KV = 6
*   KW = 9

**Sheva**


*   K = 7
*   KST = 8
*   KV = 7
*   KW = 8

**Dede**


*   K = 6
*   KST = 9
*   KV = 8
*   KW = 7

**Firda**


*   K = 9
*   KST = 6
*   KV = 7
*   KW = 8

# Library

In [ ]:
import numpy as np
import pandas as pd
from numpy.linalg import eig

In [ ]:
def calculate_priority_vector(criteria_matrix):
    eigenvalues, eigenvectors = eig(criteria_matrix)
    max_eigenvalue_index = np.argmax(eigenvalues)
    max_eigvec = eigenvectors[:, max_eigenvalue_index]
    priority_vector = max_eigvec / np.sum(max_eigvec)
    return priority_vector.real

# Perbandingan berpasangan untuk kriteria dan menghitung bobot prioritas kriteria

In [ ]:

criteria_matrix = np.array([
    [1, 2, 3, 4],
    [1/2, 1, 2, 3],
    [1/3, 1/2, 1, 2],
    [1/4, 1/3, 1/2, 1]
])

criteria_weights = calculate_priority_vector(criteria_matrix)

print("Bobot Prioritas Kriteria: ", criteria_weights)

Bobot Prioritas Kriteria:  [0.46729598 0.27718059 0.16008848 0.09543495]


In [ ]:
# Skor prioritas kriteria yang telah dihitung sebelumnya
criteria_weights = [0.46729598 ,0.27718059, 0.16008848, 0.09543495]

# Nama kriteria
criteria = ['Komunikasi', 'Kerja Sama Tim', 'Kreativitas', 'Ketepatan Waktu']

# Membuat DataFrame
df_prioritas_kriteria = pd.DataFrame({
    'Kriteria': criteria,
    'Bobot Prioritas': criteria_weights
})
print(df_prioritas_kriteria)

          Kriteria  Bobot Prioritas
0       Komunikasi         0.467296
1   Kerja Sama Tim         0.277181
2      Kreativitas         0.160088
3  Ketepatan Waktu         0.095435


# Perbandingan karyawan untuk komunikasi, KST, KV, KW

In [ ]:
communication_matrix = np.array([
    [1, 8/7, 8/6, 8/9],
    [7/8, 1, 7/6, 7/9],
    [6/8, 6/7, 1, 6/9],
    [9/8, 9/7, 9/6, 1]
])

teamwork_matrix = np.array([
    [1, 7/8, 7/9, 7/6],
    [8/7, 1, 8/9, 8/6],
    [9/7, 9/8, 1, 9/6],
    [6/7, 6/8, 6/9, 1]
])

creativity_matrix = np.array([
    [1, 6/7, 6/8, 6/7],
    [7/6, 1, 7/8, 7/7],
    [8/6, 8/7, 1, 8/7],
    [7/6, 7/8, 7/8, 1]
])

punctuality_matrix = np.array([
    [1, 9/8, 9/7, 9/8],
    [8/9, 1, 8/7, 8/8],
    [7/9, 7/8, 1, 7/8],
    [8/9, 8/7, 8/8, 1]
])


communication_weights = calculate_priority_vector(communication_matrix)
teamwork_weights = calculate_priority_vector(teamwork_matrix)
creativity_weights = calculate_priority_vector(creativity_matrix)
punctuality_weights = calculate_priority_vector(punctuality_matrix)


print("Bobot Prioritas Komunikasi: ", communication_weights)
print("Bobot Prioritas Kerja Sama Tim: ", teamwork_weights)
print("Bobot Prioritas Kreativitas: ", creativity_weights)
print("Bobot Prioritas Ketepatan Waktu: ", punctuality_weights)

Bobot Prioritas Komunikasi:  [0.26666667 0.23333333 0.2        0.3       ]
Bobot Prioritas Kerja Sama Tim:  [0.23333333 0.26666667 0.3        0.2       ]
Bobot Prioritas Kreativitas:  [0.2159865  0.25198425 0.287982   0.24404724]
Bobot Prioritas Ketepatan Waktu:  [0.2809368  0.2497216  0.2185064  0.25083519]


# Menggabungkan Hasil Final Score

In [ ]:
final_scores =(
    criteria_weights [0] * communication_weights +
    criteria_weights [1] * teamwork_weights +
    criteria_weights [2] * creativity_weights +
    criteria_weights [3] * punctuality_weights
)
print("Skor Akhir Karyawan: ", final_scores)

Skor Akhir Karyawan:  [0.25067587 0.2471225  0.24356912 0.25863251]


# Menentukan Karyawan terbaik dan terburuk

In [ ]:

karyawan = ['Nada', 'Sheva', 'Dede', 'Firda']
best_employee_index = np.argmax(final_scores)
worst_employee_index = np.argmin(final_scores)

best_employee = karyawan[best_employee_index]
worst_employee = karyawan[worst_employee_index]

print("Karyawan Terbaik Ialah: ", best_employee)
print("Karyawan Terburuk Ialah: ", worst_employee)


Karyawan Terbaik Ialah:  Firda
Karyawan Terburuk Ialah:  Dede


# Kesimpulan

Berdasarkan motode AHP, bobot priotitas kriteria untuk evaluasi karyawan menunjukkan bahwa Komunikasi(K) memiliki bobot tertinggi sebesar 46.73%, diikuti oleh Kerja Sama Tim(KST) sebesar 27.72%, Kreativitas(KV) sebesar 16.01%, dan Ketepatan Waktu(KW) sebesar 9.54%. ini juga menandakan bahwa dalam penilaian kinerja karyawan, aspek komunikasi dianggap paling krusial, sementara Ketepatan Waktu memiliki pengaru yang paling kecil. Berdasarkan bobot prioritas tersebut, evaluasi terhadap karyawan menunjukan bahwa Firda merupakan karyawan terbaik dengan skot tinggi di Komunikasi(9) dan Ketepatan Waktu(8), sedangkan Dede dianggap sebagai karyawan terburuk meskipun memiliki skor tinggi di Kerja Sama Tim(9) dan Kreativitas(8), namun skor Komunikasi(6) lebih rendah. Data ini juga dapat membantu dalam fokus evaluasi dan pengembangan terhadap karyawan serta dalam pengambilan keputusan yang lebih efektif terkait menentukan karyawan terbaik dalam satu tahun terakir berdasarkan kinerja.